### PR TITLE
Use custom function to properly parse matchspec

### PR DESCRIPTION
**Pull request by AntoinePrv**
For https://github.com/mamba-org/mamba/issues/2302

### CODE REVIEW

**Review by Klaim — APPROVED** [review]
LGTM but my understanding of the semantics of the modified functions is not completely clear so I recommend getting another approval at least.

**Review by JohanMabille — COMMENTED** [review]
Nitpicking remakrs, but overall looks good to me?

**Review by JohanMabille — APPROVED** [review]

File: libmamba/include/mamba/core/match_spec.hpp [hunk 1]
```diff
@@ -18,7 +19,7 @@ namespace mamba
     public:
 
         MatchSpec() = default;
-        MatchSpec(const std::string& i_spec);
+        MatchSpec(std::string_view i_spec);
```
[the remark this answers is not in the document]
Comment (AntoinePrv) [reply libmamba/include/mamba/core/match_spec.hpp:22]: This was the current behavior. Leaving a note to change it when this is worked on.

File: libmamba/src/core/pool.cpp [hunk 2]
```diff
@@ -148,10 +149,85 @@ namespace mamba
...
+            if ((*chan) == needle)
+            {
+                return true;
+            }
+
+            auto& custom_multichannels = Context::instance().custom_multichannels;
+            auto x = custom_multichannels.find(needle.name());
+            if (x != custom_multichannels.end())
+            {
+                for (auto el : (x->second))
```
[the remark this answers is not in the document]
Comment (AntoinePrv) [reply libmamba/src/core/pool.cpp:173]: This function was moved. I'm leaving it as it is since it needs re-work as well.

**Pull request merged**